src/services/item/plugins/geolocation/index.test.ts

```diff
@@ -0,0 +1,323 @@
+    describe('Signed out', () => {
+      it('Get geolocation for public item', async () => {
+        ({ app } = await build({ member: null }));
+        const member = await saveMember(BOB);
+        item = await savePublicItem({ actor: member, item: getDummyItem() });
+        const geoloc = await repository.save({ item, lat: 1, lng: 2, country: 'de' });
+
+        const res = await app.inject({
+          method: HttpMethod.GET,
+          url: `${ITEMS_ROUTE_PREFIX}/${item.id}/geolocation`,
+        });
+        expect(res.statusCode).toBe(StatusCodes.OK);
+        expect(res.json()).toMatchObject({
+          lat: geoloc.lat,
+          lng: geoloc.lng,
+          country: geoloc.country,
+        });
+      });
+      it('Throws for non public item', async () => {
+        ({ app } = await build({ member: null }));
+        const member = await saveMember(BOB);
+        ({ item } = await saveItemAndMembership({ member }));
+        await repository.save({ item, lat: 1, lng: 2, country: 'de' });
+
+        const res = await app.inject({
+          method: HttpMethod.GET,
+          url: `${ITEMS_ROUTE_PREFIX}/${item.id}/geolocation`,
+        });
+        expect(res.statusCode).toBe(StatusCodes.FORBIDDEN);
+        expect(res.json()).toMatchObject(new MemberCannotAccess(expect.anything()));
+      });
+    });
+    describe('Signed in', () => {
+      beforeEach(async () => {
+        ({ app, actor } = await build());
+        ({ item } = await saveItemAndMembership({ member: actor }));
+      });
+
+      it('Get geolocation', async () => {
+        const geoloc = await repository.save({ item, lat: 1, lng: 2, country: 'de' });
+        const res = await app.inject({
+          method: HttpMethod.GET,
+          url: `${ITEMS_ROUTE_PREFIX}/${item.id}/geolocation`,
+        });
+        expect(res.statusCode).toBe(StatusCodes.OK);
+        expect(res.json()).toMatchObject({
+          lat: geoloc.lat,
+          lng: geoloc.lng,
+          country: geoloc.country,
+        });
+      });
+
+      it('Get geolocation without country', async () => {
+        const geoloc = await repository.save({ item, lat: 1, lng: 2, country: null });
+        const res = await app.inject({
+          method: HttpMethod.GET,
+          url: `${ITEMS_ROUTE_PREFIX}/${item.id}/geolocation`,
+        });
+        expect(res.statusCode).toBe(StatusCodes.OK);
+        expect(res.json()).toMatchObject({
+          lat: geoloc.lat,
+          lng: geoloc.lng,
+          country: null,
+        });
+      });
+
+      it('Throws if no geolocation', async () => {
+        const res = await app.inject({
+          method: HttpMethod.GET,
+          url: `${ITEMS_ROUTE_PREFIX}/${item.id}/geolocation`,
+        });
+        expect(res.statusCode).toBe(StatusCodes.NOT_FOUND);
+        expect(res.json()).toMatchObject(new ItemGeolocationNotFound(expect.anything()));
+      });
+
+      it('Throws if id is incorrect', async () => {
```

Comment:
Maybe `Throws if id doesn't have good format` or something like that is a little bit more explicit ? It's maybe only me, but I thought it was to test an id that not exist 😅.